flutter_loggy_dio/lib/flutter_loggy_dio/dio_logger.dart

```diff
@@ -54,48 +52,41 @@ class LoggyDioInterceptor extends Interceptor with DioLoggy {
       _prettyPrintObject(options.extra, header: 'Extras');
     }
     if (requestBody && options.method != 'GET') {
-      final dynamic data = options.data;
+      final Object? data = options.data;
       if (data != null) {
         if (data is FormData) {
-          final formDataMap = <String, dynamic>{}
-            ..addEntries(data.fields)
-            ..addEntries(data.files);
-          _prettyPrintObject(formDataMap,
-              header: 'Form data | ${data.boundary}');
+          final Map<String, Object> formDataMap = <String, Object>{}..addEntries(data.fields)..addEntries(data.files);
+          _prettyPrintObject(formDataMap, header: 'Form data | ${data.boundary}');
         } else {
           _prettyPrintObject(data, header: 'Body');
         }
       }
     }
 
     _commit(requestLevel ?? LogLevel.info);
-    return options;
   }
 
   @override
-  Future onError(DioError err) async {
+  void onError(DioError err, ErrorInterceptorHandler handler) async {
     if (error) {
       if (err.type == DioErrorType.response) {
         logPrint(
-            '<<< DioError │ ${err.request?.method} │ ${err.response?.statusCode} ${err.response?.statusMessage} │ ${err.response?.request.uri.toString()}');
+            '<<< DioError │ ${err.requestOptions.method} │ ${err.response?.statusCode} ${err.response?.statusMessage} │ ${err.response?.requestOptions.uri.toString()}');
         if (err.response != null && err.response!.data != null) {
-          _prettyPrintObject(err.response!.data,
-              header: 'DioError │ ${err.type}');
+          _prettyPrintObject(err.response!.data, header: 'DioError │ ${err.type}');
```

Comment:
You should always avoid force casting, dart should have a way to safely unwrap an optional 😄